lib/vscode/src/vs/server/node/channel.ts

```diff
@@ -564,27 +565,26 @@ export class TerminalProviderChannel implements IServerChannel<RemoteAgentConnec
 			toResource: (relativePath: string) => resources.joinPath(activeWorkspaceUri, relativePath),
 		} : undefined;
 
-		const resolverService = new VariableResolverService(remoteAuthority, args, process.env as platform.IProcessEnvironment);
-		const resolver = terminalEnvironment.createVariableResolver(activeWorkspace, resolverService);
+		const resolverService = new VariableResolverService(remoteAuthority, args, process.env);
+		const resolver = terminalEnvironment.createVariableResolver(activeWorkspace, process.env, resolverService);
 
 		const getDefaultShellAndArgs = async (): Promise<{ executable: string; args: string[] | string }> => {
 			if (shellLaunchConfig.executable) {
-				const executable = resolverService.resolve(activeWorkspace, shellLaunchConfig.executable);
+				const executable = await resolverService.resolveAsync(activeWorkspace, shellLaunchConfig.executable);
 				let resolvedArgs: string[] | string = [];
 				if (shellLaunchConfig.args && Array.isArray(shellLaunchConfig.args)) {
 					for (const arg of shellLaunchConfig.args) {
-						resolvedArgs.push(resolverService.resolve(activeWorkspace, arg));
+						resolvedArgs.push(await resolverService.resolveAsync(activeWorkspace, arg));
 					}
 				} else if (shellLaunchConfig.args) {
-					resolvedArgs = resolverService.resolve(activeWorkspace, shellLaunchConfig.args);
+					resolvedArgs = await resolverService.resolveAsync(activeWorkspace, shellLaunchConfig.args);
 				}
 				return { executable, args: resolvedArgs };
 			}
 
 			const executable = terminalEnvironment.getDefaultShell(
```

Comment:
This function (and a few others in `terminalEnvironment`) are marked as deprecated, but VSCode itself continues to use it, so I figured we could too until upstream changes/removes them.